addons/base_automation/models/base_automation.py

```diff
@@ -349,6 +353,8 @@ def _compute_field_value(self, field):
                 # retrieve the action rules to possibly execute
                 actions = self.env['base.automation']._get_actions(self, ['on_write', 'on_create_or_write'])
                 records = self.filtered('id').with_env(actions.env)
+                if not actions or not records:
+                    return _compute_field_value.origin(self, field)
                 # check preconditions on records
                 pre = {action: action._filter_pre(records) for action in actions}
```

Comment:
Might be worth doing the check here (or *also* doing it here), looking at our main db about half the automated actions have a prefilter.

Reply:
Hi Xavier, thank for review.
I fixed all them except this one.

I didn't get you sorry
What is the suggested change?

Reply:
Above we're checking if there are records to perform actions on, but this bit then applies `_filter_pre` on all the records to see what actions apply to what records. It's possible that some or all actions would be mapped to no records.

Though I guess really taking advantage of this would be quite a bit more complicated as we'd need to filter this map to remove actions with no records, then probably only read records for which there are actions, then when iterating on actions check that they are still in `pre`. So forget about it, it's probably not worth it unless we know for sure that the prefilter *very commonly* needs to process nothing.